src/lib/mdlUpgrade.js

```diff
@@ -1,8 +1,45 @@
+import React from 'react';
 import { findDOMNode } from 'react-dom';
 import { mdl } from "exports-loader?mdl=componentHandler!material-design-lite/material";
 
-const mdlUpgrade = (WrappedComponent) => {
-  return class extends WrappedComponent {
+/**
+ * Returns a component class that MDL upgrades when componentDidMount() and
+ * MDL downgrades when componentWillUnmount()
+ *
+ * @param {class | function} Component - either a class extending React.Component
+ *                                       or a function.
+ */
+const mdlUpgrade = (Component) => {
```

Comment:
@rsjavier14 see here how I've added support for functional components and class components. It took me some precious time to find out I needed to merge the `defaultProps` in `_upgradeFunction()`. I also added a section about this function in the README.